pkg/estimator/server/server.go

```diff
@@ -252,18 +260,30 @@ func (es *AccurateSchedulerEstimatorServer) GetUnschedulableReplicas(ctx context
 	return &pb.UnschedulableReplicasResponse{UnschedulableReplicas: unschedulables}, err
 }
 
-func (es *AccurateSchedulerEstimatorServer) maxAvailableReplicas(nodes []*corev1.Node, request corev1.ResourceList) int32 {
-	var maxReplicas int32
-	for _, node := range nodes {
+func (es *AccurateSchedulerEstimatorServer) maxAvailableReplicas(
+	ctx context.Context,
+	ncw *nodeutil.NodeClaimWrapper,
+	nodes []*corev1.Node,
+	request corev1.ResourceList,
+) int32 {
+	var res int32
+	processNode := func(i int) {
+		node := nodes[i]
+		if node == nil {
+			return
+		}
+		if !ncw.IsNodeMatched(node) {
+			return
+		}
 		maxReplica, err := es.nodeMaxAvailableReplica(node, request)
 		if err != nil {
 			klog.Errorf("Error: %v", err)
-			continue
+			return
 		}
-		klog.V(4).Infof("Node(%s) max available replica: %d", node.Name, maxReplica)
-		maxReplicas += maxReplica
+		atomic.AddInt32(&res, maxReplica)
 	}
-	return maxReplicas
+	es.parallelizer.Until(ctx, len(nodes), processNode)
```

Comment:
why do you create the structure **NewParallelizer**， not use workqueue.ParallelizeUntil directly？